numba_cuda/numba/cuda/cudadrv/devicearray.py

```diff
@@ -96,9 +101,6 @@ def __init__(self, shape, strides, dtype, stream=0, gpu_data=None):
         self._dummy = dummy = dummyarray.Array.from_desc(
             0, shape, strides, itemsize
         )
-        # confirm that all elements of shape are ints
-        if not all(isinstance(dim, (int, np.integer)) for dim in shape):
```

Comment:
I moved this because caching avoids having to run this code every time this constructor is called, saving a good chunk of time in the constructor.